### ISSUE TITLE
Internal error when static checking with an ArrayList

### ISSUE THREAD

**Issue by slyubomirsky** [OpenJML/OpenJML#450]
Similarly to issue #260, I encountered errors when using a java ArrayList (also using the Z3 4.3.0 solver). However, the error here was not a stack overflow but rather a NullPointerException.

This following test program:

```
import java.util.ArrayList;

public class ArrList {
    public ArrayList<Integer> theList;

    //@ assignable theList;
    //@ ensures theList != null;
    //@ ensures theList.indexOf(a) == 0;
    public ArrList(int a) {
        theList = new ArrayList<Integer>();
        theList.add(a);
    }

    //@ assignable \nothing;
    public static void main(String[] args) {
        ArrList al = new ArrList(1);
        System.out.println(al.theList.get(0));
    }
}
```

resulted in the error in the attached file when running ESC:
[error.txt](https://github.com/OpenJML/OpenJML/files/350243/error.txt)

**Comment by slyubomirsky** [comment]
I also encounter an error when compiling with RAC and running, possibly related:
Exception in thread "main" java.lang.NoClassDefFoundError: org/jmlspecs/utils/Utils
    at ArrList.<init>(ArrList.java:10)
    at ArrList.main(ArrList.java:17)

**Comment by slyubomirsky** [comment]
Note: Identical errors encountered if the ArrayList is swapped for a java.util.LinkedList

**Comment by jsinglet** [comment]
Hi Steven,

Thanks for this error report. 

About the RAC error: Is jmlruntime.jar on your classpath? Perhaps try running your RAC example from the command line with a released version of OpenJML just to rule out the possibility that it's a plugin issue. 

Best,
JLS

**Comment by slyubomirsky** [comment]
Re: the RAC, I get the same error if I run from the command line using the 0.6.3 release.

**Comment by slyubomirsky** [comment]
Also perhaps informative is that we had an included test fail for similar reasons: sfbuf407 in OpenJMLTest, failing on line 11:
```
java:11: warning: The prover cannot establish an assertion (Postcondition) in method add
public void add(String s){
                  ^
java:10: warning: Associated declaration
//@ ensures (elements.contains(s));
       ^
``
This error persisted even if we added assumptions that elements and s were not null. We also encountered the same NoClassDefFoundError when attempting to execute this test file with RAC

**Comment by davidcok** [comment]
Thanks Steven - we are trying to pull together a openJML for Java 7 release and will be reviewing this problem in that context.
- David

**Comment by jsinglet** [comment]
@davidcok FYI, I've been looking at this issue today. The problem occurs in the processing of the signals clauses. Namely, on like 5377, exceptionCallDecl is initialized as follows:

```
/*@ nullable */ 
        JCVariableDecl exceptionDeclCall = 
                translatingJML && esc? null : treeutils.makeVarDef(syms.exceptionType, exceptionNameCall, methodDecl.sym, that.pos);
```
In the problematic case, we reach the `SIGNALS` case of the switch where `exceptionDeclCall` is used to make an Ident and obviously fails.

**Comment by davidcok** [comment]
If you can formulate it as a simple test case, I have an intern to look at it.
- David

**Comment by davidcok** [comment]
This test does not crash now, but does give different results in -minQuant and -no-minQuant, so leaving it open.

**Comment by davidcok** [comment]
Confirmed fixed in V0.8.19
